### DISCUSSION TITLE
How to create a full screen Windows Desktop application

### DISCUSSION THREAD

**Discussion by jldearmas** [unoplatform/uno#15008] [Q&A] [answered]
I started to test Uno Platform as an alternative to .NET MAUI, which by the way has been a bad experience to work with it.

I installed last Uno's Extension (Templates 5.x) to Visual Studio 2022 (v 17.8), run Uno.Check tool, and finally created a minimal project in order to create a simple full screen app. I only added Windows as Platform.

After searching and searching in internet I added to App class a constructor setting `PreferredLaunchWindowingMode`.

```
    public App()
    {
        ApplicationView.PreferredLaunchWindowingMode = ApplicationViewWindowingMode.FullScreen;
    }
```

But it didn't work.

![image](https://github.com/unoplatform/uno/assets/60520217/d9511d85-388c-4d2b-955e-4edcc8038957)

**Comment by limeniye** [comment]
Hello, @jldearmas.
**ApplicationView** is not supported on WinUI.
Lets check it [there](https://learn.microsoft.com/en-gb/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#core-unsupported-classes)

**Reply by limeniye** [reply, marked as answer]
try to use this way:
https://learn.microsoft.com/en-us/answers/questions/980604/winui-3-fullscreen

**Reply by jldearmas** [reply]
Thanks you so much, I'll try with that and let you know my results.

It could be nice if you (at Uno Foundation) create a new (platform independent) API to allow some functions like:

- Set Fullscreen mode
- Set Size & Position
- Set maxzimized size

Just a recomendation ;)

**Reply by limeniye** [reply]
>Just a recomendation ;)

Thank you for your recomendation. [This](https://learn.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=wasdk#title-bar-components) will be implemented in the following issues: https://github.com/unoplatform/uno/issues/10874 https://github.com/unoplatform/uno/issues/14275

I have little changed the sample and it works for me:
```C#
private AppWindow GetAppWindowForCurrentWindow()
{
    var appWindowId = XamlRoot.ContentIslandEnvironment.AppWindowId;
    return AppWindow.GetFromWindowId(appWindowId);
}
```

[UnoAppFullsizeMode.zip](https://github.com/unoplatform/uno/files/13895139/UnoAppFullsizeMode.zip)
